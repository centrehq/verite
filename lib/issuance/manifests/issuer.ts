import { CredentialIssuer } from "lib/verity"

export const manifestIssuer: CredentialIssuer = {
  id: "did:web:verity.id",
  comment: "JSON-LD definition at https://verity.id/.well_known/did.json",
  name: "Verity",
  styles: {}
}
