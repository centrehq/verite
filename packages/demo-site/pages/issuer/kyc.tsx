import IssuerLayout from "components/issuer/Layout"
import { currentUser, requireAuth } from "lib/auth-fns"
import { temporaryAuthToken } from "lib/database"
import type { User } from "lib/database"
import { NextPage } from "next"
import QRCode from "qrcode.react"
import { ManifestUrlWrapper } from "types"

type Props = {
  manifestUrlWrapper: ManifestUrlWrapper
  user: User
}

export const getServerSideProps = requireAuth<Props>(async (context) => {
  const user = await currentUser(context)
  const authToken = await temporaryAuthToken(user)
  const manifestUrlWrapper: ManifestUrlWrapper = {
    manifestUrl: `${process.env.HOST}/api/issuance/manifests/kyc`,
    submissionUrl: `${process.env.HOST}/api/issuance/submission/${authToken}`,
    version: "1"
  }

  return {
    props: {
      manifestUrlWrapper,
      user
    }
  }
})

const KycAmlPage: NextPage<Props> = ({ manifestUrlWrapper, user }) => {
  const stats = [
    { name: "Jumio Score", stat: user.jumioScore },
    { name: "OFAC Score", stat: user.ofacScore }
  ]

  return (
    <IssuerLayout title="KYC/AML Attestation">
      <div className="flex flex-col justify-center space-y-8">
        <dl className="flex flex-row mx-auto space-x-2 sm:space-x-5">
          {stats.map((item) => (
            <div
              key={item.name}
              className="px-4 py-3 overflow-hidden text-center bg-white rounded-lg shadow sm:py-5 sm:px-6 sm:px-8 flex-0"
            >
              <dt className="text-sm font-medium text-gray-500 truncate">
                {item.name}
              </dt>
              <dd className="mt-1 text-3xl font-semibold text-gray-900">
                {item.stat}
              </dd>
            </div>
          ))}
        </dl>
        <QRCode
          value={JSON.stringify(manifestUrlWrapper)}
          className="w-48 h-48 mx-auto"
          renderAs="svg"
        />
        <textarea
          className="container h-40 mx-auto font-mono text-sm border-2"
          readOnly
          value={JSON.stringify(manifestUrlWrapper, null, 4)}
        />
      </div>
    </IssuerLayout>
  )
}

export default KycAmlPage
