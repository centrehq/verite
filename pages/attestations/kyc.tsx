import { GetServerSideProps, NextPage } from "next"
import { getSession } from "next-auth/client"
import QRCode from "qrcode.react"
import Authenticated, { SessionProps } from "components/Authenticated"
import Layout from "components/Layout"
import { findUser, temporaryAuthToken, User } from "lib/database"
import { ManifestUrlWrapper } from "types"

type Props = SessionProps & {
  manifestUrlWrapper: ManifestUrlWrapper
  user: User
}

export const getServerSideProps: GetServerSideProps<Props> = async (
  context
) => {
  const session = await getSession(context)

  if (!session || !session.user) {
    return {
      redirect: {
        destination: "/",
        permanent: false
      }
    }
  }

  const user = findUser((session.user as User).id)
  const authToken = await temporaryAuthToken(user)
  const manifestUrlWrapper: ManifestUrlWrapper = {
    manifestUrl: `${process.env.HOST}/api/issuance/manifests/kyc`,
    submissionUrl: `${process.env.HOST}/api/issuance/submission/${authToken}`,
    version: "1"
  }

  return {
    props: {
      manifestUrlWrapper,
      session,
      user
    }
  }
}

const KycAmlPage: NextPage<Props> = ({ manifestUrlWrapper, user }) => {
  const stats = [
    { name: "Jumio Score", stat: user.jumioScore },
    { name: "OFAC Score", stat: user.ofacScore }
  ]

  return (
    <Authenticated>
      <Layout title="KYC/AML Attestation">
        <div className="flex flex-col justify-center space-y-8">
          <dl className="flex flex-row mx-auto space-x-5">
            {stats.map((item) => (
              <div
                key={item.name}
                className="px-6 py-5 overflow-hidden text-center bg-white rounded-lg shadow sm:px-8 flex-0"
              >
                <dt className="text-sm font-medium text-gray-500 truncate">
                  {item.name}
                </dt>
                <dd className="mt-1 text-3xl font-semibold text-gray-900">
                  {item.stat}
                </dd>
              </div>
            ))}
          </dl>
          <QRCode
            value={JSON.stringify(manifestUrlWrapper)}
            className="w-48 h-48 mx-auto"
            renderAs="svg"
          />
          <textarea
            className="container h-40 mx-auto font-mono text-sm border-2"
            readOnly
            value={JSON.stringify(manifestUrlWrapper, null, 4)}
          />
        </div>
      </Layout>
    </Authenticated>
  )
}

export default KycAmlPage
