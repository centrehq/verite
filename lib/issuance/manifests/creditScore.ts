import { manifestIssuer } from "./issuer"
import { createCreditScoreManifest, CredentialManifest } from "lib/verity"

export const creditScoreManifest: CredentialManifest =
  createCreditScoreManifest(manifestIssuer, {
    thumbnail: {
      uri: "https://verity.id/img/logo.png",
      alt: "Verity Logo"
    },
    hero: {
      uri: "https://verity.id/img/kycCred.png",
      alt: "KYC Visual"
    },
    background: {
      color: "#ff0000"
    },
    text: {
      color: "#d4d400"
    }
  })
