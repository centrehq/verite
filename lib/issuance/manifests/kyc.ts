import { manifestIssuer } from "./issuer"
import { createKycAmlManifest, CredentialManifest } from "lib/verity"

export const kycManifest: CredentialManifest = createKycAmlManifest(
  manifestIssuer,
  {
    thumbnail: {
      uri: "https://verity.id/img/logo.png",
      alt: "Verity Logo"
    },
    hero: {
      uri: "https://verity.id/img/kycCred.png",
      alt: "KYC Visual"
    },
    background: {
      color: "#ff0000"
    },
    text: {
      color: "#d4d400"
    }
  }
)
