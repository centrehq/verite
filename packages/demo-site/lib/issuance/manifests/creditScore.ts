import { createCreditScoreManifest, CredentialManifest } from "verity"
import { manifestIssuer } from "./issuer"

export const creditScoreManifest: CredentialManifest =
  createCreditScoreManifest(manifestIssuer, {
    thumbnail: {
      uri: `${process.env.HOST}/img/logo.png`,
      alt: "Verity Logo"
    },
    hero: {
      uri: `${process.env.HOST}/img/credit-score-hero.png`,
      alt: "Credit Score Visual"
    },
    background: {
      color: "#ff0000"
    },
    text: {
      color: "#d4d400"
    }
  })
