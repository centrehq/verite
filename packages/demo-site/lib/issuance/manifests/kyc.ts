import { createKycAmlManifest, CredentialManifest } from "verity"
import { manifestIssuer } from "./issuer"

export const kycManifest: CredentialManifest = createKycAmlManifest(
  manifestIssuer,
  {
    thumbnail: {
      uri: `${process.env.HOST}/img/logo.png`,
      alt: "Verity Logo"
    },
    hero: {
      uri: `${process.env.HOST}/img/kyc-aml-hero.png`,
      alt: "KYC+AML Visual"
    },
    background: {
      color: "#ff0000"
    },
    text: {
      color: "#d4d400"
    }
  }
)
