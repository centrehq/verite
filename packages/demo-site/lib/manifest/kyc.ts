import { buildKycAmlManifest } from "@centre/verity"
import type { CredentialManifest } from "@centre/verity"
import { fullURL } from "../utils"
import { manifestIssuer } from "./issuer"

export const kycManifest: CredentialManifest = buildKycAmlManifest(
  manifestIssuer,
  {
    thumbnail: {
      uri: fullURL("/img/kyc-aml-thumbnail.png"),
      alt: "Verity Logo"
    },
    hero: {
      uri: fullURL("/img/kyc-aml-hero.png"),
      alt: "KYC+AML Visual"
    },
    background: {
      color: "#EC4899"
    },
    text: {
      color: "#FFFFFF"
    }
  }
)
