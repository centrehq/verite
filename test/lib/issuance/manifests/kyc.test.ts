import { kycManifest } from "lib/issuance/manifests/kyc"

describe("Tests KYC Manifest", () => {
  it("tests for breaking KYC Manifest schema changes", async () => {
    expect.assertions(1)
    const km = kycManifest
    expect(km).toMatchSnapshot()
  })
})
