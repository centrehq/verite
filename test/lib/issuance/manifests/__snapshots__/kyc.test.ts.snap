// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests KYC Manifest tests for breaking KYC Manifest schema changes 1`] = `
Object {
  "format": Object {
    "jwt_vc": Object {
      "alg": Array [
        "EdDSA",
        "ES256K",
      ],
    },
    "jwt_vp": Object {
      "alg": Array [
        "EdDSA",
        "ES256K",
      ],
    },
  },
  "id": "KYCAMLAttestation",
  "issuer": Object {
    "comment": "JSON-LD definition at https://verity.id/.well_known/did.json",
    "id": "did:web:verity.id",
    "name": "Verity",
    "styles": Object {},
  },
  "output_descriptors": Array [
    Object {
      "description": "Attestation that Verity has completed KYC/AML verification for this subject",
      "display": Object {
        "description": Object {
          "text": "The KYC authority processes Know Your Customer and Anti-Money Laundering analysis, potentially employing a number of internal and external vendor providers.",
        },
        "properties": Array [
          Object {
            "label": "Authority",
            "path": Array [
              "$.KYCAMLAttestation.authorityName",
            ],
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "label": "Authority URL",
            "path": Array [
              "$.KYCAMLAttestation.authorityUrl",
            ],
            "schema": Object {
              "format": "uri",
              "type": "string",
            },
          },
          Object {
            "label": "Approved At",
            "path": Array [
              "$.KYCAMLAttestation.approvalDate",
            ],
            "schema": Object {
              "format": "date-time",
              "type": "string",
            },
          },
        ],
        "subtitle": Object {
          "fallback": "Includes date of approval",
          "path": Array [
            "$.approvalDate",
            "$.vc.approvalDate",
          ],
        },
        "title": Object {
          "fallback": "Verity KYC Attestation",
          "path": Array [
            "$.KYCAMLAttestation.authorityName",
          ],
        },
      },
      "id": "kycAttestationOutput",
      "name": "Proof of KYC from Verity",
      "schema": Array [
        Object {
          "uri": "https://verity.id/schemas/identity/1.0.0/KYCAMLAttestation",
        },
      ],
      "styles": Object {
        "background": Object {
          "color": "#ff0000",
        },
        "hero": Object {
          "alt": "KYC+AML Visual",
          "uri": "http://localhost:3000/img/kyc-aml-hero.png",
        },
        "text": Object {
          "color": "#d4d400",
        },
        "thumbnail": Object {
          "alt": "Verity Logo",
          "uri": "http://localhost:3000/img/logo.png",
        },
      },
    },
  ],
  "presentation_definition": Object {
    "format": Object {
      "jwt_vp": Object {
        "alg": Array [
          "EdDSA",
          "ES256K",
        ],
      },
    },
    "id": "ProofOfControlPresentationDefinition",
    "input_descriptors": Array [
      Object {
        "id": "proofOfIdentifierControlVP",
        "name": "Proof of Control Verifiable Presentation",
        "purpose": "A Verifiable Presentation establishing proof of identifier control over the DID.",
        "schema": Array [
          Object {
            "uri": "/.well-known/verifiablePresentationSchema.json",
          },
        ],
      },
    ],
  },
  "version": "0.1.0",
}
`;
